spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -75,10 +75,15 @@ public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String a
         log.debug("The gateway with the ID {} requests its latest API key list.", gatewayIdAndApiKey);
         try {
             Optional<Proxy> proxy = Optional.ofNullable(proxyRepository.findById(gatewayIdAndApiKey));
-            proxy.get().updateTimeOfLastSuccessfulCallToAdmin_setToNow();
-            proxyRepository.save(proxy.get());
-            return proxy.map(value -> new ResponseEntity<>(getApiKeysBelongingToServicesOf(value), HttpStatus.OK))
-                    .orElseGet(() -> new ResponseEntity<>(null, HttpStatus.NOT_FOUND));
+
+            if (proxy.isPresent()) {
+                proxy.get().updateTimeOfLastSuccessfulCallToAdmin_setToNow();
```

Comment:
Instead of calling `.get()` three times, I would suggest to use a local variable to improve readability.

Reply:
I created the getProxyFromOptional as localvariable

Reply:
That name is confusing and hard to read, why not `proxyToUpdate`?

Reply:
Done